### PR TITLE
Build objective-c examples as part of automatic tests

### PR DESCRIPTION
**Pull request by y-zeng**
Related issue: #3371

### CODE REVIEW

File: tools/run_tests/run_tests.py [hunk 1]
```diff
@@ -425,7 +425,7 @@ def make_options(self):
     return []
 
   def build_steps(self):
-    return [['tools/run_tests/build_python.sh', tox_env] 
+    return [['tools/run_tests/build_python.sh', tox_env]
```
Comment (makdharma) [inline tools/run_tests/run_tests.py]: Why is this change needed in python? Shouldn't this PR be only about ObjC?

Comment (y-zeng) [inline tools/run_tests/run_tests.py]: Reverted. Thanks a lot for the review! It seems my auto format tool deleted this blank.

**Comment by jtattermusch** [comment]
Sure, trying to build examples is a good thing to do, but against what version of gRPC are the examples going to run?  Do they pick up the commit currently being tested or will they get built using some release tag? (if they do, these tests become more confusing than useful and it's going to lead to trouble)

**Comment by jcanizales** [comment]
The examples should always get gRPC from the Cocoapods repo. And every time we push a new version we should then modify them to use that one.

So if I break gRPC, the examples keep building but the other tests will fail. If I break the examples (or a new version of XCode does), the test will tell me.

The grey area here is protoc and the plugin, because right now the examples are using whichever version is installed in your machine. #4518 will fix that.

File: src/objective-c/tests/build_one_example.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+# SCHEME - scheme of the example, used by xcodebuild
+
+# CocoaPods requires the terminal to be using UTF-8 encoding.
+export LANG=en_US.UTF-8
+
+cd `dirname $0`/../../..
+
+cd $EXAMPLE_PATH
+
+# clean the directory
```
Comment (jcanizales) [inline src/objective-c/tests/build_one_example.sh]: Is this to remove `Podfile.lock` and the XCode workspace?

Comment (y-zeng) [inline src/objective-c/tests/build_one_example.sh]: Yes, I was trying to remove files generated by `pod install` and revert its change to `project.pbxproj`.

Comment (jcanizales) [inline src/objective-c/tests/build_one_example.sh]: If one's made changes to the example and hasn't committed them, will this revert the changes?

Comment (y-zeng) [inline src/objective-c/tests/build_one_example.sh]: If the changes are not staged, running this test will revert them. If this is too aggressive, it might be good to only remove the files ignored by git.

Comment (jcanizales) [inline src/objective-c/tests/build_one_example.sh]: Yeah let's just do `rm -rf Pods; rm Podfile.lock; rm -rf $SCHEME.xcworkspace`, with whatever option `rm` uses to not complain if the file's not found. That has less risk of someone cursing us in the future :)

The changes in the project file should be fine to check in.

Comment (y-zeng) [inline src/objective-c/tests/build_one_example.sh]: Fixed. Thanks a lot for the review!

File: src/objective-c/tests/build_example_test.sh [hunk 3]
```diff
@@ -35,22 +35,35 @@ set -eo pipefail
...
-EXAMPLE_PATH=examples/objective-c/helloworld \
-SCHEME=HelloWorld                            \
-./build_one_example.sh
+BINDIR=`pwd`/../../../bins/$CONFIG
+TMP_PATH=$PATH
+hash protoc 2>/dev/null || TMP_PATH=$BINDIR/protobuf:$TMP_PATH
+PATH=$TMP_PATH hash protoc-gen-objcgrpc 2>/dev/null || {
+  ln -sf $BINDIR/grpc_objective_c_plugin $BINDIR/protoc-gen-objcgrpc
+  TMP_PATH=$BINDIR:$TMP_PATH
+}
```
Comment (jcanizales) [inline src/objective-c/tests/build_example_test.sh]: Please add a comment above this section explaining what it's doing.

Comment (y-zeng) [inline src/objective-c/tests/build_example_test.sh]: Done. Thanks for the review!

**Comment by jcanizales** [comment]
LGTM

**Comment by jtattermusch** [comment]
I don't understand how you can mark the PR as "ready to merge" when objc tests haven't finished yet. Looking at the diffs, this change definitely has potential to break things.

**Comment by jcanizales** [comment]
Sorry for the confusion. The reason is because Travis isn't running `run_tests.py`, so it's unaffected by this change.
